services/src/main/java/edu/unc/lib/dl/cdr/services/processing/RunEnhancementsService.java

```diff
@@ -0,0 +1,132 @@
+
+import java.util.HashMap;
+import java.util.ArrayList;
+import java.util.List;
+
+import edu.unc.lib.dl.acl.service.AccessControlService;
+import edu.unc.lib.dl.acl.util.AgentPrincipals;
+import edu.unc.lib.dl.acl.util.Permission;
+import edu.unc.lib.dl.fcrepo4.PIDs;
+import edu.unc.lib.dl.fedora.PID;
+import edu.unc.lib.dl.metrics.TimerFactory;
+import edu.unc.lib.dl.model.DatastreamPids;
+import edu.unc.lib.dl.search.solr.model.BriefObjectMetadata;
+import edu.unc.lib.dl.search.solr.model.Datastream;
+import edu.unc.lib.dl.search.solr.model.SearchRequest;
+import edu.unc.lib.dl.search.solr.model.SearchResultResponse;
+import edu.unc.lib.dl.search.solr.model.SearchState;
+import edu.unc.lib.dl.search.solr.model.SimpleIdRequest;
+import edu.unc.lib.dl.services.AbstractMessageSender;
+import edu.unc.lib.dl.ui.service.SolrQueryLayerService;
+
+public class RunEnhancementsService extends AbstractMessageSender {
+    private static final Logger LOG = LoggerFactory.getLogger(RunEnhancementsService.class);
+    private static final Timer timer = TimerFactory.createTimerForClass(RunEnhancementsService.class);
+
+    private AccessControlService aclService;
+
+    @Autowired
+    protected SolrQueryLayerService queryLayer;
+    private Datastream recordInfo;
+
+    public void run(AgentPrincipals agent, ArrayList<HashMap> objectPids, Boolean force) {
+        try (Timer.Context context = timer.time()) {
+            for (HashMap objectPid : objectPids) {
+                String uuid = (String) objectPid.get("pid");
+                PID pid = PIDs.get(uuid);
+
+                aclService.assertHasAccess("User does not have permission to run enhancements",
+                        pid, agent.getPrincipals(), Permission.runEnhancements);
+
+                LOG.debug("sending solr update message for {} of type {}", pid, "runEnhancements");
+
+                String objectType = (String) objectPid.get("objectType");
+
+                if (!objectType.equals("File")) {
+                    SearchRequest searchRequest = new SearchRequest();
+                    searchRequest.setAccessGroups(agent.getPrincipals());
+                    searchRequest.setSearchState(new SearchState());
+                    searchRequest.setRootPid(pid);
+                    searchRequest.setApplyCutoffs(false);
+                    SearchResultResponse resultResponse = queryLayer.performSearch(searchRequest);
+
+                    for (BriefObjectMetadata metadata : resultResponse.getResultList()) {
+                        createMessage(metadata, pid, agent.getUsername(), force);
+                    }
+                } else {
+                    SimpleIdRequest searchRequest = new SimpleIdRequest(uuid, agent.getPrincipals());
+                    BriefObjectMetadata metadata = queryLayer.getObjectById(searchRequest);
+                    createMessage(metadata, pid, agent.getUsername(), force);
+                }
+            }
+        } catch (Exception e) {
```

Comment:
Hmm, I think you can remove this catch and leave it up to the thing that calls it to decide whether or not to log the error. Similarly, you might want to add some logging to the controller to log the error in the catch block it already has.